core/src/main/scala/cats/mtl/KindTransformer.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats
+package mtl
+
+import cats.arrow.FunctionK
+import cats.data.EitherT
+import cats.data.IorT
+import cats.data.Kleisli
+import cats.data.OptionT
+import cats.data.StateT
+import cats.~>
+
+/**
+ * A utility for transforming the higher-kinded type `F` to another higher-kinded type `G`.
+ */
+@annotation.implicitNotFound("No transformer defined from ${F} to ${G}")
+trait KindTransformer[F[_], G[_]] {
+
+  /**
+   * A higher-kinded function that lifts the kind `F` into a `G`.
+   *
+   * @note
+   *   This method is usually best implemented by a `liftK` method on `G`'s companion object.
+   */
+  val liftK: F ~> G
+
+  /**
+   * Modify the context of `G[A]` using the natural transformation `f`.
+   *
+   * This method is "limited" in the sense that while most `mapK` methods can modify the context
+   * using arbitrary transformations, this method can only modify the context using
+   * transformations from `F` to `F`.
+   *
+   * @note
+   *   This method is usually best implemented by a `mapK` method on `G`.
+   */
+  def limitedMapK[A](ga: G[A])(f: F ~> F): G[A]
+
+  /**
+   * Lifts a natural transformation from `F` to `F` into a natural transformation from `G` to
+   * `G`.
+   *
+   * @note
+   *   Implementors SHOULD NOT override this method; the only reason it is not final is for
+   *   optimization of the identity case.
+   */
+  def liftFunctionK(f: F ~> F): G ~> G =
+    new (G ~> G) {
+      def apply[A](ga: G[A]): G[A] = limitedMapK(ga)(f)
+    }
```

Comment:
for simplicity, you can think of `limitedMapK` as a specialisation/optimisation of `liftFunctionK`; the purpose of `KindTransformer` is to provide both `F ~> G` and `(F ~> F) => (G ~> G)`